engine/execution/state/cargo/queue/queue.go

```diff
@@ -0,0 +1,115 @@
+package queue
+
+import (
+	"sync"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type headerInContext struct {
+	Header *flow.Header
+	ID     flow.Identifier
+}
+
+// FinalizedBlockQueue is a concurrency-safe queue of finalized block headers
+// Every time a new header is added to the queue it verifies the complience of the header to the
+// most recenlty added header.
+// Under the hood a circular buffer with a limited capacity is used
+type FinalizedBlockQueue struct {
```

Comment:
your queue implementation seems overly complicated.  the dumb slice "queue" will probably perform equally well with minimum memory penalty.

If you feel strongly about using a circular buffer queue, then please refactor it into a generic non-theadsafe implementation, and have FinalizedBlockQueue use that.

Reply:
moved to the slice one for now and removed capacity for now.